scripts/update_secrets.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+)
+
+// Check if all the rules that exist in "gitleaks" are included in our list of rules (in secret.go file)
+func main() {
+
+	//Find the latest release of "gitleaks"
+	release, err := fetchLatestRelease()
+	if err != nil {
+		fmt.Printf("Error fetching latest release: %s\n", err)
+		return
+	}
+
+	// Import the rules from "gitleaks"
+	rawURL := fmt.Sprintf("https://raw.githubusercontent.com/zricethezav/gitleaks/%s/cmd/generate/config/main.go", release.TagName)
+	content, err := fetchRemoteContent(rawURL)
+	if err != nil {
+		fmt.Printf("Error fetching remote content: %s\n", err)
+		return
+	}
+	reGitleaks := regexp.MustCompile(`configRules\s*=\s*append\(configRules,\s*rules\.([a-zA-Z0-9_]+)\(`)
+	matchesGitleaks := reGitleaks.FindAllStringSubmatch(string(content), -1)
+
+	//Import the rules from our project "2ms"
+	localContent, err := fetchLocalContent("secrets/secrets.go")
+	if err != nil {
+		fmt.Printf("Error fetching local content: %s\n", err)
+		return
+	}
+	reLocal := regexp.MustCompile(`allRules\s*=\s*append\(allRules,\s*Rule{Rule:\s*\*rules\.([a-zA-Z0-9_]+)\(\),`)
+	matchLocal := reLocal.FindAllStringSubmatch(string(localContent), -1)
+
+	//Insert the rules in map for good run time in search
+	localRulesMap := make(map[string]bool)
+	for _, match := range matchLocal {
+		localRulesMap[match[1]] = true
+	}
+
+	// Compare the rules and check if missing rules in our list of rules
+	missingInLocal := []string{}
+	for _, rule := range matchesGitleaks {
+		if _, found := localRulesMap[rule[1]]; !found {
+			missingInLocal = append(missingInLocal, rule[1])
+		}
+	}
+
+	if len(missingInLocal) > 0 {
+		fmt.Printf("%d differences between our rules and the rules in the new version in 'gitleaks' were found: \n \n", len(missingInLocal))
```

Comment:
Add the version number to the message, and maybe add the link to the `main` file in _gitleaks_ in a new line, why not :-)